src/Module/Contacts.php

```diff
@@ -0,0 +1,1137 @@
+		foreach ($orig_records as $orig_record) {
+			$contact_id = $orig_record['id'];
+			if (x($_POST, 'contacts_batch_update')) {
+				self::_contact_update($contact_id);
+				$count_actions++;
+			}
+			if (x($_POST, 'contacts_batch_block')) {
+				self::_contact_block($contact_id);
+				$count_actions++;
+			}
+			if (x($_POST, 'contacts_batch_ignore')) {
+				self::_contact_ignore($contact_id);
+				$count_actions++;
+			}
+			if (x($_POST, 'contacts_batch_archive')) {
+				$r = self::_contact_archive($contact_id, $orig_record);
+				if ($r) {
+					$count_actions++;
+				}
+			}
+			if (x($_POST, 'contacts_batch_drop')) {
+				self::_contact_drop($orig_record);
+				$count_actions++;
+			}
+		}
+		if ($count_actions > 0) {
+			info(L10n::tt("%d contact edited.", "%d contacts edited.", $count_actions));
+		}
+
+		goaway('contacts');
+	}
+
+	public static function post()
+	{
+		$a = self::getApp();
+
+		if (!local_user()) {
+			return;
+		}
+
+		if ($a->argv[1] === "batch") {
+			self::contacts_batch_actions($a);
+			return;
+		}
+
+		$contact_id = intval($a->argv[1]);
+		if (!$contact_id) {
+			return;
+		}
+
+		if (!DBA::exists('contact', ['id' => $contact_id, 'uid' => local_user()])) {
+			notice(L10n::t('Could not access contact record.') . EOL);
+			goaway('contacts');
+			return; // NOTREACHED
+		}
+
+		Addon::callHooks('contact_edit_post', $_POST);
+
+		$profile_id = intval(defaults($_POST, 'profile-assign', 0));
+		if ($profile_id) {
+			if (!DBA::exists('profile', ['id' => $profile_id, 'uid' => local_user()])) {
+				notice(L10n::t('Could not locate selected profile.') . EOL);
+				return;
+			}
+		}
+
+		$hidden = intval($_POST['hidden']);
+
+		$notify = intval($_POST['notify']);
+
+		$fetch_further_information = intval(defaults($_POST, 'fetch_further_information', 0));
+
+		$ffi_keyword_blacklist = escape_tags(trim(defaults($_POST, 'ffi_keyword_blacklist', '')));
+
+		$priority = intval(defaults($_POST, 'poll', 0));
+		if ($priority > 5 || $priority < 0) {
+			$priority = 0;
+		}
+
+		$info = escape_tags(trim($_POST['info']));
+
+		$r = q("UPDATE `contact` SET `profile-id` = %d, `priority` = %d , `info` = '%s',
+			`hidden` = %d, `notify_new_posts` = %d, `fetch_further_information` = %d,
+			`ffi_keyword_blacklist` = '%s' WHERE `id` = %d AND `uid` = %d",
+			intval($profile_id),
+			intval($priority),
+			DBA::escape($info),
+			intval($hidden),
+			intval($notify),
+			intval($fetch_further_information),
+			DBA::escape($ffi_keyword_blacklist),
+			intval($contact_id),
+			intval(local_user())
+		);
+		if (DBA::isResult($r)) {
+			info(L10n::t('Contact updated.') . EOL);
+		} else {
+			notice(L10n::t('Failed to update contact record.') . EOL);
+		}
+
+		$contact = DBA::selectFirst('contact', [], ['id' => $contact_id, 'uid' => local_user()]);
+		if (DBA::isResult($contact)) {
+			$a->data['contact'] = $contact;
+		}
+
+		return;
+	}
+
+	/* contact actions */
+
+	private static function _contact_update($contact_id)
```

Comment:
According to the PSR-2 coding standards, class methods and properties name shouldn't start with an underscore to denote privacy scope. Please rename to `updateContactFromPoll`.